src/lib/components/tabsv2/Tabsv2.component.js

```diff
@@ -0,0 +1,402 @@
+//@flow
+import React, {
+  createContext,
+  useCallback,
+  useRef,
+  useMemo,
+  useEffect,
+  useState,
+} from 'react';
+import type { Element, ChildrenArray } from 'react';
+import {
+  TabBar,
+  ScrollableContainer,
+  TabContent,
+  TabItem,
+  TabsContainer,
+  TabsScroller,
+} from './StyledTabs';
+import { useHistory, useLocation } from 'react-router-dom';
+import { matchPath, Route, Switch } from 'react-router';
+import { SecondaryText, BasicText, EmphaseText } from '../text/Text.component';
+import { animate } from './animate';
+import ScrollButton from './ScrollButton';
+import Tab from './Tab';
+import type { TabProps } from './Tab';
+
+const nextItem = (list, item) => {
+  if (list === item) {
+    return list.firstChild;
+  }
+  if (item && item.nextElementSibling) {
+    return item.nextElementSibling;
+  }
+  return list.firstChild;
+};
+
+const previousItem = (list, item) => {
+  if (list === item) {
+    return list.lastChild;
+  }
+  if (item && item.previousElementSibling) {
+    return item.previousElementSibling;
+  }
+  return list.lastChild;
+};
+
+const moveFocus = (list, currentFocus, traversalFunction) => {
+  let wrappedOnce = false;
+  let nextFocus = traversalFunction(list, currentFocus);
+
+  while (nextFocus) {
+    // Prevent infinite loop.
+    if (nextFocus === list.firstChild) {
+      if (wrappedOnce) {
+        return;
+      }
+      wrappedOnce = true;
+    }
+
+    if (!nextFocus.hasAttribute('tabindex')) {
+      // Move to the next element.
+      nextFocus = traversalFunction(list, nextFocus);
+    } else {
+      nextFocus.focus();
+      return;
+    }
+  }
+};
+
+type TabsProps = {
+  activeTabColor?: string,
+  activeTabSeparator?: string,
+  tabLineColor?: string,
+  inactiveTabColor?: string,
+  tabContentColor?: string,
+  separatorColor?: string,
+  tabHoverColor?: string,
+  children: ChildrenArray<Element<typeof Tab>>,
+  className?: string,
+};
+
+export const TabsContext = createContext<boolean>(false);
+
+function Tabs({
+  activeTabColor,
+  activeTabSeparator,
+  tabLineColor,
+  inactiveTabColor,
+  tabContentColor,
+  separatorColor,
+  tabHoverColor,
+  children,
+  className,
+  ...rest
+}: TabsProps) {
+  const [selectedTabIndex, setSelectedTabIndex] = useState(null);
+  const location = useLocation();
+  const history = useHistory();
+  const [displayScroll, setDisplayScroll] = useState({
+    start: false,
+    end: false,
+  });
+
+  const scrollButtonEndRef = useRef<?HTMLDivElement>(null);
+  const scrollButtonStartRef = useRef<?HTMLDivElement>(null);
+  const tabsListRef = useRef<any>(null);
+  const tabsRef = useRef<any>(null);
+
+  const scroll = (scrollValue: number, { animation = true } = {}) => {
```

Comment:
This function is never called with `animation = false` so I assume the parameter and corresponding branch can be removed as they are never used nor tested.